### PR TITLE
Implement FreeBSD support in various projects

### PR DESCRIPTION
**Pull request by josteink**
SecureString and System.Diagnostics.Debug.

CC: @janhenke @stephentoub

### CODE REVIEW

File: src/System.Diagnostics.Debug/src/System.Diagnostics.Debug.csproj [hunk 1]
```diff
@@ -22,6 +22,8 @@
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'OSX_Release|AnyCPU' " />
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Windows_Debug|AnyCPU' " />
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Windows_Release|AnyCPU' " />
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'FreeBSD_Debug|AnyCPU' " />
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'FreeBSD_Release|AnyCPU' " />
```
Comment (janhenke) [inline src/System.Diagnostics.Debug/src/System.Diagnostics.Debug.csproj]: A nit: If we want to keep alphabetic ordering, FreeBSD should come first before Linux.

File: src/System.Diagnostics.Debug/src/System.Diagnostics.Debug.csproj [hunk 2]
```diff
@@ -98,6 +100,15 @@
...
   </ItemGroup>
+  <!-- FreeBSD -->
+  <ItemGroup Condition="'$(TargetsFreeBSD)' == 'true'">
+    <Compile Include="$(CommonPath)\Interop\FreeBSD\Interop.Errors.cs">
+      <Link>Common\Interop\FreeBSD\Interop.Errors.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\FreeBSD\libc\Interop.OpenFlags.cs">
+      <Link>Common\Interop\FreeBSD\Interop.OpenFlags.cs</Link>
+    </Compile>
+  </ItemGroup>
```
Comment (janhenke) [inline src/System.Diagnostics.Debug/src/System.Diagnostics.Debug.csproj]: Same here: Nit, keep alphabetic order.

File: src/System.Security.SecureString/src/System.Security.SecureString.csproj [hunk 3]
```diff
@@ -18,6 +18,8 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'OSX_Release|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Windows_Debug|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Windows_Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'FreeBSD_Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'FreeBSD_Release|AnyCPU'" />
```
Comment (janhenke) [inline src/System.Security.SecureString/src/System.Security.SecureString.csproj]: Alphabetic order (nit).

File: src/System.Security.SecureString/src/System.Security.SecureString.csproj [hunk 4]
```diff
@@ -95,9 +97,17 @@
...
       <Link>Common\Interop\OSX\Interop.SysConfNames.cs</Link>
     </Compile>
+  <ItemGroup Condition=" '$(TargetsFreeBSD)' == 'true' ">
+    <Compile Include="$(CommonPath)\Interop\FreeBSD\libc\Interop.MemoryMappedFlags.cs">
+      <Link>Common\Interop\FreeBSD\Interop.MemoryMappedFlags.cs</Link>
+    </Compile>
+    <Compile Include="$(CommonPath)\Interop\FreeBSD\libc\Interop.SysConfNames.cs">
+      <Link>Common\Interop\FreeBSD\Interop.SysConfNames.cs</Link>
+    </Compile>
+  </ItemGroup>
```
Comment (janhenke) [inline src/System.Security.SecureString/src/System.Security.SecureString.csproj]: Last alphabetic order nit. ;)

**Comment by Maxwe11** [comment]
@dotnet-bot test this please

**Comment by josteink** [comment]
Nits addressed. Let's hope the CI shapes up eventually and gets this green :)

**Comment by stephentoub** [comment]
LGTM

**Pull request merged**